install_config/aggregate_logging.adoc

```diff
@@ -3628,3 +3636,304 @@ cluster. Edit your non-cluster logging service (for example, `logging-es`,
 ----
 $ oc patch svc/logging-es -p '{"spec":{"selector":{"component":"es","provider":"openshift"}}}'
 ----
+
+[[optimal-state-cluster]]
+=== Cluster not in an optimal state
+
+The reasons that your cluster may not be considered to be in an optimal state are
+that the cluster health is not `green` and/or the cluster size specified in the
+cluster health does not match the number of pods running for that cluster (e.g.
+not all nodes have joined the cluster).
+
+To check which currently applies to your cluster:
+
+. Get the name of all pods for your particular cluster
++
+----
+$ oc get pods -l component=es[-ops] -o name
+----
+
+. Check the health of the cluster using one of the pods found above
+----
+$ oc exec -c elasticsearch <es_pod> -- es_cluster_health
+----
+
+The total number of nodes in the cluster displayed from the second command should
+match the number of pods returned in the first command.
+
+==== If the cluster is not `green`
+
+If the cluster is currently in a red state, see the section xref:recover-yellow-state[unable to recover to yellow].
```

Comment:
Would it make sense to bump the "unable to recover to yellow" and "unable to recover to green" sections here (instead of linking) and adjust the headings accordingly?

[[if-the-cluster-is-not-green]]
=== If the Cluster is Not `Green`

[[recover-yellow-state]]
==== Unable to Recover to Yellow State 
[content]

[[recover-green-state]]
==== Unable to Recover to Green State
[content]

Reply:
Would it make sense then to also bump `If the Cluster Size Does Not Match Pod Count`?

My original thought was that since a user would be directed here because their cluster was not in an optimal state, it made sense to have these two points as sub headers . However, it makes sense that we would just have the `recover-yellow-state` and `recover-green-state` headers underneath the cluster not being green as you suggested... should we just nest those two underneath as you're suggesting, but not bump them up at all? I'm open to formatting suggestions

Reply:
@ewolinetz 
I like the idea of nesting those two sections. Not sure what you mean by "not bump them up at all." I don't like losing the headings from the TOC at the top of the file (makes them easier to find). 

[[if-the-cluster-is-not-green]]
=== If the Cluster is Not Green

[[recover-yellow-state]]
==== Unable to Recover to Yellow State
[content]

[[recover-green-state]]
==== Unable to Recover to Green State
[content]
[[if-the-cluster-size-does-not-match-pod-count]]
==== If the Cluster Size Does Not Match Pod Count

Reply:
Just curious. How would a user know the cluster is not optimal? I find it unusual that the first step in the section is how to figure out why it is not optimal. ES indicates it is not optimal, but not why, without you digging for the reason?

Reply:
@mburke5678 I didn't realize how out of context that statement was. 

The idea of it not being optimal comes from output that the ansible role will provide [1]. So the thought is that initially a user would come to this page and look for that header when encountering the message from the logging role. And perhaps as the phrase is used more readily it becomes synonymous with the cluster not being `green` and having the intended number of members.

[1] https://github.com/openshift/openshift-ansible/pull/8415/files#diff-523f0217fa77f7435b018e6eda535463R46

Reply:
I went ahead and nested those two sections similar to what you had pasted (the exception being that the "If the Cluster Size Does Not Match Pod Count" header I bumped up "===" instead of "===="

Reply:
@ewolinetz Looks good. Probably should drop the two _If the Cluster_ topics back to ==== (subordinate to Cluster Not in an Optimal State) to really drive the synonymity you mention. But, I like them in the TOC.
Shall we leave this to @bmcelvee to decide? or does this need to get in ASAP?

Reply:
I think I'm liking them appearing in the ToC as well.. We can let Brandi make the final judgement call on this though. This is attached to ansible work that was made available in 3.10, but I don't think its so critical that it can't wait for her sign off. I'll squash my commits in the mean time.